src/groups/mqb/mqbc/mqbc_storageutil.cpp

```diff
@@ -2700,8 +2700,57 @@ void StorageUtil::forceFlushFileStores(FileStores* fileStores)
     }
 }
 
+void StorageUtil::purgeDomainDispatched(
+    bsl::vector<bsl::vector<mqbcmd::PurgeQueueResult> >* purgedQueuesVec,
+    bslmt::Latch*                                        latch,
+    int                                                  partitionId,
+    const FileStores&                                    fileStores,
+    const bsl::string&                                   domainName)
+{
+    // executed by *QUEUE_DISPATCHER* thread with the specified 'partitionId'
+
+    // PRECONDITIONS
+    BSLS_ASSERT_SAFE(fileStores[partitionId]->inDispatcherThread());
+    BSLS_ASSERT_SAFE(latch);
+    BSLS_ASSERT_SAFE(purgedQueuesVec);
+    BSLS_ASSERT_SAFE(0 <= partitionId);
+    BSLS_ASSERT_SAFE(static_cast<unsigned int>(partitionId) <
+                     fileStores.size());
+    BSLS_ASSERT_SAFE(purgedQueuesVec->size() == fileStores.size());
+
+    bsl::vector<mqbcmd::PurgeQueueResult>& purgedQueues =
+        (*purgedQueuesVec)[partitionId];
+    fileStores[partitionId]->purgeDomain(&purgedQueues, domainName);
+
+    latch->arrive();
+}
+
+void StorageUtil::purgeQueueDispatched(
+    mqbcmd::PurgeQueueResult* purgedQueue,
+    bslmt::Semaphore*         purgeFinishedSemaphore,
+    mqbs::FileStore*          fileStore,
```

Comment:
I think this arg is still needed, because:
1. `fileStore` contains info about its dispatcher thread, which is necessary for thread safety checks. `mqbi::Storage` doesn't have this info.
2. `fileStore` contains info about its primary node. Again, `mqbi::Storage` doesn't have this information.